src/app/Recordings/ActiveRecordingsTable.tsx

```diff
@@ -227,50 +229,78 @@ export const ActiveRecordingsList: React.FunctionComponent<ActiveRecordingsListP
         return (<span>{str}</span>);
       };
 
-      return <>
-        <DataListCell key={`table-row-${props.index}-1`}>
-          {props.recording.name}
-        </DataListCell>
-        <DataListCell key={`table-row-${props.index}-2`}>
-          <ISOTime timeStr={props.recording.startTime} />
-        </DataListCell>
-        <DataListCell key={`table-row-${props.index}-3`}>
-          <RecordingDuration duration={props.recording.duration} />
-        </DataListCell>
-        <DataListCell key={`table-row-${props.index}-4`}>
-          {props.recording.state}
-        </DataListCell>
-      </>
+      return (
+        <Tr key={`${props.index}_parent`}>
+          <Td key={`active-table-row-${props.index}_0`}>
+            <Checkbox
+              name={`active-table-row-${props.index}-check`}
+              onChange={handleCheck}
+              isChecked={checkedIndices.includes(props.index)}
+              id={`active-table-row-${props.index}-check`}
+            />
+          </Td>
+          <Td
+            key={`active-table-row-${props.index}_1`}
+            id={`active-ex-toggle-${props.index}`}
+            aria-controls={`active-ex-expand-${props.index}`}
+            expand={{
+              rowIndex: props.index,
+              isExpanded: isExpanded,
+              onToggle: handleToggle,
+            }}
+          />
+          <Td key={`active-table-row-${props.index}_2`} dataLabel={tableColumns[0]}>
+            {props.recording.name}
+          </Td>
+          <Td key={`active-table-row-${props.index}_3`} dataLabel={tableColumns[1]}>
+            <ISOTime timeStr={props.recording.startTime} />
+          </Td>
+          <Td key={`active-table-row-${props.index}_4`} dataLabel={tableColumns[2]}>
+            <RecordingDuration duration={props.recording.duration} />
+          </Td>
+          <Td key={`active-table-row-${props.index}_5`} dataLabel={tableColumns[3]}>
+            {props.recording.state}
+          </Td>
+          <RecordingActions
+            index={props.index}
+            recording={props.recording}
+            uploadFn={() => context.api.uploadActiveRecordingToGrafana(props.recording.name)}
+          />
+        </Tr>
+      )
+    }, [props.recording, props.recording.name, props.duration, props.index]);
```

Comment:
I think the memo dependencies array here should also contain `handleCheck`, `checkedIndices`, `props.index`, `isExpanded`, `handleToggle`, `tableColumns`, `props.recording.startTime`, `props.recording.duration`, `props.recording.state`, and `context.api`

Reply:
Sounds good, I'll add these in.

For the case of `props.recording.whatever`, do they need to be separate entries into the dependencies array? Or would `props.recording` suffice?

Reply:
I think in practice, `props.recording` would be enough - however this could be broken if the `recording` prop passed to the child component is updated in-place by mutating its state without changing the reference of the `recording` object. It's probably safest to specifically list out each of the properties of the `recording` object. Otherwise, if the parent component mutates the recording object without replacing it, these effects in the child component won't re-run/memos won't recalculate/etc.

Reply:
Ah right, sounds good